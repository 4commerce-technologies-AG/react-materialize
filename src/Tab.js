import React from 'react';
import PropTypes from 'prop-types';
import Col from './Col';

// This is just a holder for the props and children for tab, thus
// there is no logic here.
const Tab = ({ children, idx, className }) => (
  <Col id={`tab_${idx}`} s={12} className={className}>
    {children}
  </Col>
);

Tab.propTypes = {
  title: PropTypes.string.isRequired,
  idx: PropTypes.string,
  children: PropTypes.node,
  className: PropTypes.string,
  active: PropTypes.bool,
  disabled: PropTypes.bool,
  tabWidth: PropTypes.number
};

export default Tab;
