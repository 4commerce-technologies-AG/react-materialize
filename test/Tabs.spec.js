import React from 'react';
import { shallow, mount } from 'enzyme';
import Tabs from '../src/Tabs';
import Tab from '../src/Tab';
import mocker from './helper/new-mocker';

describe('Tabs', () => {
  let wrapper;
  const tabsInitMock = jest.fn();
  const tabInstanceDestroyMock = jest.fn();
  const tabsMock = {
    init: (el, options) => {
      tabsInitMock(options);
      return {
        destroy: tabInstanceDestroyMock
      };
    }
  };
  const restore = mocker('Tabs', tabsMock);

  const options = {
    duration: 400,
    onShow: jest.fn(),
    swipeable: true,
    responsiveThreshold: 2
  };

  wrapper = shallow(
    <Tabs>
      <Tab title="one">One</Tab>
      <Tab title="Two">Two</Tab>
    </Tabs>
  );

  afterAll(() => {
    restore();
  });

  beforeEach(() => {
    tabsInitMock.mockClear();
  });

  test('should create list of Tab itemt', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('with options', () => {
    beforeEach(() => {
      wrapper = mount(
        <Tabs tabOptions={options}>
          <Tab title="one">One</Tab>
          <Tab title="Two">Two</Tab>
        </Tabs>
      );
    });

    test('initializes Tabs with options', () => {
      expect(tabsInitMock).toHaveBeenCalledWith(options);
    });
  });

  describe('when updated', () => {
    beforeEach(() => {
      wrapper = mount(
        <Tabs tabOptions={options}>
          <Tab title="one">One</Tab>
          <Tab title="Two">Two</Tab>
        </Tabs>
      );
    });

    test('should re-initialize with options', () => {
      expect(tabsInitMock).toHaveBeenCalledWith(options);
      tabsInitMock.mockClear();
      tabInstanceDestroyMock.mockClear();
      wrapper.setProps({ className: 'test' });
      expect(tabInstanceDestroyMock).toHaveBeenCalled();
      expect(tabsInitMock).toHaveBeenCalledWith(options);
    });
  });

  describe('when unmounted', () => {
    beforeEach(() => {
      tabInstanceDestroyMock.mockClear();
      wrapper.unmount();
    });

    test('should be destroyed', () => {
      expect(tabInstanceDestroyMock).toHaveBeenCalled();
    });
  });
});
