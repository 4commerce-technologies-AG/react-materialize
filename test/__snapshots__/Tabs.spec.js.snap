// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should create list of Tab itemt 1`] = `
<Row>
  <Col
    s={12}
  >
    <ul
      className="tabs"
    >
      <li
        className="tab col"
        key="mockId0/.0"
      >
        <a
          className=""
          href="#tab_mockId0"
          onClick={[Function]}
        >
          one
        </a>
      </li>
      <li
        className="tab col"
        key="mockId1/.1"
      >
        <a
          className=""
          href="#tab_mockId1"
          onClick={[Function]}
        >
          Two
        </a>
      </li>
    </ul>
  </Col>
  <Col
    id="tab_mockId0"
    key="tabmockId0/.0"
    s={12}
    style={
      Object {
        "display": "none",
      }
    }
  >
    One
  </Col>
  <Col
    id="tab_mockId1"
    key="tabmockId1/.1"
    s={12}
    style={
      Object {
        "display": "none",
      }
    }
  >
    Two
  </Col>
</Row>
`;
